include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -436,10 +446,36 @@ struct Session : Base<OrtSession> {
   size_t GetOutputCount() const;                  ///< Returns the number of model outputs
   size_t GetOverridableInitializerCount() const;  ///< Returns the number of inputs that have defaults that can be overridden
 
+  // [[deprecated]] 
```

Comment:
Is the [[deprecated]] just to mark it as such in doxygen? I thought you might be using the [[deprecated]] C++ attribute

Reply:
We are not removing and I do not want stop build warnings for places we use it. We would need to refactor next time around when we remove it.

Reply:
Yeah, we can't remove it, but we could strongly encourage anyone with the new headers to switch away with the warning. It's tricky though, as you don't want to instantly break builds. I don't know the right answer, and I don't think we have a policy on this yet.